source/common/io/io_uring.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
```

Comment:
This interface is tied tightly to syscalls which are specific to Linux and not generic enough. I made it abstract to ease unit testing of the io_uring-based iohandle (coming in followups).
But given [similar syscalls](https://windows-internals.com/i-o-rings-when-one-i-o-operation-is-not-enough/) are developed for Windows it might be useful to make it more generic. The problem is that I don't know how to design a proper abstraction at this stage.

Reply:
IIUC #19082 does exactly that - an IoHandle as a boostrap extension. This PR is an attempt to enable io_uring also for file IO. Additionally Matt [proposed](https://github.com/envoyproxy/envoy/pull/19082#issuecomment-981977371) to switch to io_uring whenever it's available without enabling it explicitly as an extension.